### PR TITLE
Logo generator

### PR DESCRIPTION
**Pull request by JasaNiklanovic**

### CODE REVIEW

File: logo-generator/src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,215 @@
+import React, { Component } from 'react';
+import UploadButton from 'react-upload-button-v2';
+import LogoTools from './components/LogoTools';
+import Logo from './components/Logo';
+import CodeExport from './components/CodeExport';
+//import './App.css';
```
Comment (vuknje) [inline logo-generator/src/App.jsx]: https://vignette.wikia.nocookie.net/dragonvale/images/e/e7/Tmp_13976-images.duckduckgo.com%283%29-1938063952.gif/revision/latest?cb=20151226232040

File: logo-generator/src/App.jsx [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+  updateLogo = (type, value) => {
+    let logos = this.state.logos;
+    let selectedLogo = logos.find((logo) => logo.handle === this.state.selectedLogoHandle);
+    selectedLogo[type] = value;
+    this.setState({logos: logos})
+  }
+
+  findSelectedLogo = () => {
+    return this.state.logos.find((logo) => logo.handle === this.state.selectedLogoHandle)
+  };
```
Comment (vuknje) [inline logo-generator/src/App.jsx]: this `;` screwed my syntax highlighting

File: logo-generator/src/registerServiceWorker.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
+// In production, we register a service worker to serve assets from local cache.
```
Comment (vuknje) [inline logo-generator/src/registerServiceWorker.js]: Do we need this?

File: logo-generator/src/components/LogoTools.jsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                min={20}
+                max={100}
+              />
+            </div>
+          </div>
+          <div className="navigation-tools">
+            <button onClick={() => this.props.reorderLogo('left')} className="btn"><img src={leftIcon} /></button>
+            <button onClick={() => this.props.reorderLogo('right')} className="btn"><img src={rightIcon} /></button>
+            <button onClick={() => this.props.removeLogo()} className="btn btn-secondary"><img src={crossIcon} /></button>
+            <button onClick={() => this.props.closeTools()} className="btn btn-tertiary small-text">DONE</button>
```
[the remark this answers is not in the document]
Comment (JasaNiklanovic) [reply logo-generator/src/components/LogoTools.jsx]: I agree, lets leave like this, more consistent in this particular case.

File: logo-generator/src/registerServiceWorker.js (hunk 3 above)
Comment (JasaNiklanovic) [reply logo-generator/src/registerServiceWorker.js]: I don't think we actually need this here, good call. I've removed it.

File: logo-generator/src/components/LogoTools.jsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+  onPaddingChange = (value) => {
+    this.props.updateLogo('padding', value)
+  }
+
+  onSizeChange = (value) => {
+    this.props.updateLogo('size', value)
+  }
+
+  render() {
```
[the remark this answers is not in the document]
Comment (JasaNiklanovic) [reply logo-generator/src/components/LogoTools.jsx]: Done.

File: logo-generator/src/App.jsx (hunk 2 above)
Comment (JasaNiklanovic) [reply logo-generator/src/App.jsx]: Done.

File: logo-generator/src/App.jsx (hunk 1 above)
Comment (JasaNiklanovic) [reply logo-generator/src/App.jsx]: Done.

File: logo-generator/src/components/LogoTools.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (JasaNiklanovic) [reply logo-generator/src/components/LogoTools.jsx]: Done.

File: logo-generator/src/components/CodeExport.jsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { formatPadding, formatSize } from '../helpers/helpers.js'
+
+export default class CodeExport extends React.Component {
+
+  compile = () => {
+    const { logos } = this.props;
+    let text = ``;
+    logos.forEach((logo) => {
+      text += `<div><span class='vertical-helper'></span><a href='${logo.link}'><img style='height:${formatSize(logo.size, this.props.ctaHeight)}; padding: ${formatPadding(logo.padding)}' src='${logo.url}' /></a></div>`;
```
[the remark this answers is not in the document]
Comment (JasaNiklanovic) [reply logo-generator/src/components/CodeExport.jsx]: Done.

File: logo-generator/src/App.jsx [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+      })
+      this.setState({selectedLogoHandle: null, logos: logos})
+    }
+  }
+
+  closeTools = () => {
+    this.setState({selectedLogoHandle: null})
+  }
+
+  render() {
```
[the remark this answers is not in the document]
Comment (JasaNiklanovic) [reply logo-generator/src/App.jsx]: Done.

File: logo-generator/src/App.jsx [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+        //   link: "http://google.com",
+        //   padding: 0,
+        //   size: 100
+        // },{
+        //   handle: "352345",
+        //   url: "https://cdn.fs.turtl.co/JdnesGhoS2avBHgt1Vto",
+        //   link: "http://google.com",
+        //   padding: 0,
+        //   size: 100
+        // }
```
[the remark this answers is not in the document]
Comment (JasaNiklanovic) [reply logo-generator/src/App.jsx]: I was using it for testing but ok, lets delete it.

File: logo-generator/src/App.jsx [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+  filestackKey: 'AWJxmnYWQMCMoFrRXTULjz',
+}
+
+class App extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      // potentially extract ctaHeight (in em) to input, 
+      // we need a fixed height for proper pdf logo rendering.
```
[the remark this answers is not in the document]
Comment (JasaNiklanovic) [reply logo-generator/src/App.jsx]: I updated the comment.

File: logo-generator/src/App.jsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (vuknje) [reply logo-generator/src/App.jsx]: @JasaNiklanovic there are other ways to do this without using commented code. If you think it's still valuable to have it, maybe we can have a dev endpoint or something

File: logo-generator/src/App.jsx [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+        selectedLogo = this.findSelectedLogo(),
+        selectedLogoPosition = logos.indexOf(selectedLogo),
+        moveTo = (direction === 'left') ? selectedLogoPosition - 1 : selectedLogoPosition + 1;
+
+    logos.splice(moveTo, 0, logos.splice(selectedLogoPosition, 1)[0]);
+    this.setState({logos: logos})
+  }
+
+  removeLogo = () => {
+    var r = window.confirm("Are you sure?");
```
[the remark this answers is not in the document]
Comment (JasaNiklanovic) [reply logo-generator/src/App.jsx]: Done.

File: logo-generator/src/App.jsx [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+    if (r === true) {
```
[the remark this answers is not in the document]
Comment (JasaNiklanovic) [reply logo-generator/src/App.jsx]: Done.

File: logo-generator/src/components/CodeExport.jsx [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+    return text
+  }
+
+  renderCompiled = () => {
+    return (
+      `   <script data-name="safe" >\n`  + 
+      `   angular.module("turtl.viewer")\n`  + 
+      `     .run(["$templateCache", function($templateCache) {\n`  + 
+      `       var html = $templateCache.get("turtl.viewer/toolbar.html");\n`  + 
+      `       var obj = $(html);\n`  +
```
[the remark this answers is not in the document]
Comment (JasaNiklanovic) [reply logo-generator/src/components/CodeExport.jsx]: I replaced this with htmlEl, to keep the jQuery -> El standard we have.

File: logo-generator/src/components/CodeExport.jsx [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+      `     }]);\n`  + 
+      `   </script>` +
+      `   <style>` +
+      `     .cta-logo-generator {background-image: none!important;width: auto!important; height: ${this.props.ctaHeight}em!important}` +
+      `     .cta-logo-generator div {display: inline-block; height: 100%; vertical-align: middle; text-align: center;}` +
+      `     .cta-logo-generator .vertical-helper {display: inline-block; height: 100%; vertical-align: middle;}` +
+      `     .cta-logo-generator a {border: none!important;}` +
+      `     .cta-logo-generator img {vertical-align: middle; height: 100%; border: none!important;}` +
+      `     flipbook toolbar {background: ${this.props.previewBackgroundColor}}` +
+      `   </style>`
```
[the remark this answers is not in the document]
Comment (JasaNiklanovic) [reply logo-generator/src/components/CodeExport.jsx]: Done.

File: logo-generator/src/App.jsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (JasaNiklanovic) [reply logo-generator/src/App.jsx]: Its fine actually, now that we persist the state to the localstorage its easier to keep testing data, don't need mocks anymore.

**Pull request merged**